### PR TITLE
Signature helper

### PR DESCRIPTION
**Pull request by kohr-h**
Utility for more extensible and robust printing of signatures, plus application to some cases in `discr` and `partition`. Quite some improvements in the printed strings IMO.

### CODE REVIEW

File: odl/discr/lp_discr.py [hunk 1]
```diff
@@ -440,63 +447,61 @@ def __repr__(self):
...
-            if not all(s == 'nearest' for s in self.interp_by_axis):
-                arg_fstr += ', interp={interp!r}'
-            if impl != 'numpy':
-                arg_fstr += ', impl={impl!r}'
-            if self.order != 'C':
-                arg_fstr += ', order={order!r}'
-            if nodes_on_bdry:
-                arg_fstr += ', nodes_on_bdry={nodes_on_bdry!r}'
+        if self.ndim <= 3:
+            default_ax_lbl = tuple(['$x$', '$y$', '$z$'][:self.ndim])
```
Comment (kohr-h) [inline odl/discr/lp_discr.py]: It's just two places, keeping it for now.

**Comment by kohr-h** [comment]
Updated - merge?

**Comment by adler-j** [comment]
Go ahead

**Comment by kohr-h** [comment]
> Go ahead

I can't - you need to hit the "Approve" button. (I changed the repo setting to require review and CI pass before merge. We don't want buttons hit accidentally again.)

**Comment by adler-j** [comment]
> I can't - you need to hit the "Approve" button. (I changed the repo setting to require review and CI pass before merge. We don't want buttons hit accidentally again.)

Done!

**Comment by kohr-h** [comment]
> Done!

Thx! Just one rebase and CI from merging.

**Comment by adler-j** [comment]
Merge?

**Pull request merged**